### PR TITLE
Fix issue 24686: SumType stopped working with unmatched DeducedParame…

### PR DESCRIPTION
**Pull request by ljmf00**
…terType template

### CODE REVIEW

**Comment by ljmf00** [comment]
This should probably go to stable CC @JohanEngelen @RazvanN7 @dkorpel

**Comment by ljmf00** [comment]
I took the freedom to create `Industry` label. This is blocking Weka on upgrading to the latest language version.

**Comment by ljmf00** [comment]
Buildkite error seems unrelated?

**Review by JohanEngelen — APPROVED** [review]

**Comment by thewilsonator** [comment]
please retitle the _commit message_ : "Fix issue 24686:...". also please target stable

**Comment by ljmf00** [comment]
> please retitle the _commit message_ : "Fix issue 24686:...". also please target stable

The commit message is like that. Did I miss something? I thought that was for the bot to grab.

**Comment by thewilsonator** [comment]
Gah, sorry. `"Fix _Bugzilla 24686:...".

File: std/traits.d [hunk 1]
```diff
@@ -9261,6 +9265,7 @@ package(std) template DeducedParameterType(T)
     }
 
     static assert(is(DeducedParameterType!NoCopy == NoCopy));
+    static assert(is(DeducedParameterType!(inout(NoCopy)) == inout(NoCopy)));
```
Comment (pbackus) [inline std/traits.d:9268]: This test case passes already on Phobos `master`. What's the purpose of including it in this PR?

Comment (ljmf00) [reply std/traits.d:9268]: To ensure `DeducedParameterType(alias T)` is not added directly to `DeducedParameterTypeImpl`, which makes this false, a unwanted behavior.

**Review by pbackus — APPROVED** [review]

**Comment by ljmf00-wekaio** [comment]
> Gah, sorry. `"Fix _Bugzilla_ issue 24686:...".

Got it, I certainly didn't contribute for a while, I didn't know about this change. Thanks!

**Comment by thewilsonator** [comment]
> Got it, I certainly didn't contribute for a while, I didn't know about this change. Thanks!

Well the bot's message does say that, but I didn't notice it the first few times. I only really look for the "⚠️⚠️⚠️ Warnings ⚠️⚠️", and don't read the actual content :)

**Comment by JohanEngelen** [comment]
I rebased and retargeted this PR onto stable

**Comment by JohanEngelen** [comment]
Again an unrelated CI failure... please merge this.

**Comment by ljmf00** [comment]
> I rebased and retargeted this PR onto stable

Thanks. Sorry, I'm not used to go to my personal github that often. I wish I could merge it to keep track of things more often. Need to change my workflow a bit to be aware of these kind of things.

**Pull request merged**